src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs

```diff
@@ -458,6 +458,38 @@ private static IResourceBuilder<ProjectResource> WithProjectDefaults(this IResou
                     },
                     createIfNotExists: true);
                 }
+
+                // Update URLs for endpoints from the launch profile if a launchUrl is set
+                if (Uri.TryCreate(launchProfile.LaunchUrl, UriKind.RelativeOrAbsolute, out var launchUri))
+                {
+                    builder.WithUrls(context =>
+                    {
+                        if (context.Resource.TryGetEndpoints(out var endpoints))
+                        {
+                            foreach (var endpoint in endpoints)
+                            {
+                                if (endpoint.FromLaunchProfile)
+                                {
+                                    var url = context.Urls.FirstOrDefault(u => string.Equals(u.Endpoint?.EndpointName, endpoint.Name, StringComparisons.EndpointAnnotationName));
+                                    if (url is not null)
+                                    {
+                                        if (launchUri.IsAbsoluteUri)
```

Comment:
Follow up: Discuss if an absolute launch urls wins over all over urls and becomes the only ones that shows up.